lib/events.js

```diff
@@ -1,18 +1,60 @@
+var util = require('./util');
+var nacl = require('tweetnacl');
+
+function encrypt(pusher, channel, data) {
+  if (pusher.config.encryptionMasterKey === undefined) {
+    throw new Error("Set encryptionMasterKey before triggering events on encrypted channels");
+  }
+
+  const nonceBytes = nacl.randomBytes(24);
+
+  const ciphertextBytes = nacl.secretbox(
+    Buffer.from(JSON.stringify(data), 'utf8'),
+    nonceBytes,
+    pusher.channelSharedSecret(channel));
+
+  return JSON.stringify({
+    nonce: Buffer(nonceBytes).toString('base64'),
+    ciphertext: Buffer(ciphertextBytes).toString('base64')
+  });
+}
+
 exports.trigger = function(pusher, channels, eventName, data, socketId, callback) {
-  var event = {
-    "name": eventName,
-    "data": ensureJSON(data),
-    "channels": channels
-  };
-  if (socketId) {
-    event.socket_id = socketId;
+  if (channels.length === 1 && util.isEncryptedChannel(channels[0])) {
+    var channel = channels[0];
+    var event = {
+      "name": eventName,
+      "data": encrypt(pusher, channel, data),
+      "channels": [channel]
+    };
+    if (socketId) {
+      event.socket_id = socketId;
+    }
+    pusher.post({ path: '/events', body: event }, callback);
+  } else {
+    for (var i = 0; i < channels.length; i++) {
+      if (util.isEncryptedChannel(channels[i])) {
+        throw new Error("You cannot trigger to multiple channels when using encrypted channels");
```

Comment:
It's not clear to me why this restriction needs to be in place. Could you describe the motivation in a comment?

Reply:
@jameshfisher this is the only comment I'd like to get your input on. Other points are minor, that I'm happy to tidy up.

Reply:
Documented the rationale in https://github.com/pusher/pusher-http-node/pull/86/commits/d697baf428105c31198da8bb207448e16e3171c6